java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -191,6 +192,15 @@ Object applyTransform(Transform transform, Object document ) {
                 .map(JsonPath::compile)
                 .collect(Collectors.toList()));
 
+        if (transform.getApplyOnlyWhen() != null) {
+            Object filterResult = JsonPath.compile(transform.getApplyOnlyWhen()).read(document);
+
+            ArrayList<?> results = (ArrayList<?>) filterResult;
+            if (results != null && results.size() == 0) {
+                return document;
```

Comment:
style-wise, I prefer avoiding multiple paths that `return` within a method.

i think more readable to more explicitly separate what you're doing by either having a `boolean transformApplies` that you compute the value of, or by splitting into two methods: 1) the method that actually applies the transformation, and 2) a method that evaluates the condition, and either returns directly or returns result of calling method (1).